langchain/chains/priority_chain.py

```diff
@@ -0,0 +1,133 @@
+"""Chain pipeline where the outputs of one step feed directly into next."""
+import asyncio
+import time
+from typing import Any, Dict, List, Optional, Tuple
+
+from pydantic import Extra, root_validator
+
+from langchain.callbacks.manager import (
+    AsyncCallbackManagerForChainRun,
+    CallbackManagerForChainRun,
+)
+from langchain.chains.base import Chain
+
+
+class PriorityChain(Chain):
+    """Chain where multiple chains are executed in a particular order
+    if the preferred chains time out."""
+
+    rankings: List[Dict[int, Tuple[Chain, float]]]
```

Comment:
this seems like a funky data structure, could maybe benefit from a named tuple or something

Reply:
sg, changing this now